mlrun/utils/helpers.py

```diff
@@ -1015,16 +1015,35 @@ def create_class(pkg_class: str):
     return class_
 
 
-def create_function(pkg_func: str):
+def create_function(pkg_func: str, reload_modules: bool = False):
     """Create a function from a package.module.function string
 
     :param pkg_func:  full function location,
                       e.g. "sklearn.feature_selection.f_classif"
+    :param reload_modules: reload the function if it is in the modules list.
     """
     splits = pkg_func.split(".")
     pkg_module = ".".join(splits[:-1])
     cb_fname = splits[-1]
+    pkg_module_exists_in_modules = pkg_module in sys.modules
     pkg_module = __import__(pkg_module, fromlist=[cb_fname])
+
+    if reload_modules:
+        # if the function appears in the modules list, we need to reload the code again because it may have changed
+        if pkg_module_exists_in_modules:
+            logger.warning(
+                "Reloading module. Not all associated modules can be reloaded, import them manually."
+                "Or, with Jupyter, restart the Python kernel."
+            )
```

Comment:
what kind of modules can't be reloaded? feels like some inside knowledge may be worth adding as a comment.

Reply:
I was added a recursive method, so now it will reload the module recursively and it should fix this issue - https://iguazio.atlassian.net/browse/ML-3943#:~:text=Proposition%3A%C2%A0In,will%20be%20partial.